src/EnergyPlus/ConvectionCoefficients.cc

```diff
@@ -4079,6 +4080,115 @@ void CalcISO15099WindowIntConvCoeff(EnergyPlusData &state,
         state.dataHeatBalSurf->SurfHConvInt(SurfNum) = state.dataHeatBal->LowHConvLimit;
 }
 
+RoofGeoCharacteristicsStruct getRoofGeometryInformation(EnergyPlusData &state)
+{
+    auto &Surface(state.dataSurface->Surface);
+
+    RoofGeoCharacteristicsStruct RoofGeo;
+
+    std::vector<Vector> uniqueRoofVertices;
+    std::vector<SurfaceGeometry::EdgeOfSurf> uniqEdgeOfSurfs; // I'm only partially using this
+    for (const auto &surface : Surface) {
+        if (surface.Tilt < 45.0) { // TODO Double check tilt wrt outside vs inside?
+            auto const &vertices = surface.Vertex;
+
+            auto const &thisArea = surface.Area;
+            Real64 const z_min(minval(vertices, &Vector::z));
+            Real64 const z_max(maxval(vertices, &Vector::z));
+            Real64 const verticalHeight = z_max - z_min;
+            RoofGeo.Height += verticalHeight * thisArea;
+            RoofGeo.Tilt += surface.Tilt * thisArea;
+            RoofGeo.Azimuth += surface.Azimuth * thisArea;
+            RoofGeo.Area += thisArea;
```

Comment:
Part One: figure out area and area-weighted averages of some characteristics: height, azimuth (and tilt, but that's not used)